app/validators/base_regulation_validator.rb

```diff
@@ -20,26 +20,144 @@ class BaseRegulationValidator < TradeTariffBackend::Validator
+        :explicit_abrogation_regulation_id,
+        :explicit_abrogation_regulation_role
+      ]
+
+      record.column_changes.keys.all? do |key|
+        allowed_fields_to_change.include?(key)
+      end
+    end
+  end
+
   # TODO: ROIMB7
-  # TODO: ROIMB48
-  # TODO: ROIMB44
+
+  validation :ROIMB48,
+             'If the regulation has not been abrogated, the effective end date must be greater than or equal to the base regulation end date if it is explicit.',
+             on: [:create, :update] do |record|
+    return true if record.complete_abrogation_regulation.present? && record.explicit_abrogation_regulation.present?
+
+    if record.effective_end_date.present? && record.validity_end_date.present?
+      record.effective_end_date >= record.validity_end_date
+    end
+  end
+
+  validation :ROIMB44,
+             'The "Regulation Approved Flag" indicates for a draft regulation whether the draft is approved, i.e. the regulation is definitive apart from its publication (only the definitive regulation id and the O.J. reference are not yet known). A draft regulation (regulation id starts with a \'C\') can have its "Regulation Approved Flag" set to 0=\'Not Approved\' or 1=\'Approved\'. Its flag can only change from 0=\'Not Approved\' to 1=\'Approved\'. Any other regulation must have its "Regulation Approved Flag" set to 1=\'Approved\'.' do |record|
+    if record.base_regulation_id.starts_with?("C")
+      !record.column_changed?(:approved_flag) || (record.column_changes[:approved_flag][0] == false && record.column_changes[:approved_flag][1] == true)
+    else
+      record.approved_flag
+    end
+  end
+
   # TODO: ROIMB46
-  # TODO: ROIMB47
-  # TODO: ROIMB8
+
+  validation :ROIMB47,
+             'The validity period of the regulation group id must span the validity period of the base regulation.',
+             on: [:create, :update] do |record|
+    if record.regulation_group.present?
+      record.regulation_group.validity_start_date <= record.validity_start_date &&
+        ((record.regulation_group.validity_end_date.blank? || record.validity_end_date.blank?) ||
+          (record.regulation_group.validity_end_date >= record.validity_end_date))
+    end
+  end
+
+  validation :ROIMB8,
+             'Explicit dates of related measures must be within the validity period of the base regulation.' do |record|
+    record.generating_measures.all? do |measure|
+      record.validity_start_date <= measure.validity_start_date &&
+      ((record.validity_end_date.blank? || measure.validity_end_date.blank?) ||
+        (record.validity_end_date >= measure.validity_end_date))
+    end
+  end
+
   # TODO: ROIMB9
   # TODO: ROIMB10
-  # TODO: ROIMB11
+
+  validation :ROIMB11,
+             'The start date of the first PTS in time must be within the validity period of the base regulation. Apply the rule when changing the start date further forward in time.' do |record|
+    record.generating_measures.all? do |measure|
```

Comment:
Also should we include the measure_partial_temporary_stops in the join for performance?

Reply:
I will check.